src/audio_worklet.js

```diff
@@ -32,51 +32,119 @@ function createWasmAudioWorkletProcessor(audioParams) {
       this.callback = {{{ makeDynCall('iipipipp', 'opts.callback') }}};
       this.userData = opts.userData;
       // Then the samples per channel to process, fixed for the lifetime of the
-      // context that created this processor. Note for when moving to Web Audio
-      // 1.1: the typed array passed to process() should be the same size as this
-      // 'render quantum size', and this exercise of passing in the value
-      // shouldn't be required (to be verified)
+      // context that created this processor. Even though this 'render quantum
+      // size' is fixed at 128 samples in the 1.0 spec, it will be variable in
+      // the 1.1 spec. It's passed in now, just to prove it's settable, but will
+      // eventually be a property of the  AudioWorkletGlobalScope (globalThis).
       this.samplesPerChannel = opts.samplesPerChannel;
+      this.bytesPerChannel = this.samplesPerChannel * {{{ getNativeTypeSize('float') }}};
+
+      // Creates the output views (see createOutputViews() docs)
+      this.maxBuffers = Math.min(((wwParams.stackSize - /*minimum alloc*/ 16) / this.bytesPerChannel) | 0, /*sensible limit*/ 64);
+      this.outputViews = [];
+#if ASSERTIONS
+      console.assert(this.maxBuffers > 0, `AudioWorklet needs more stack allocating (at least ${this.bytesPerChannel})`);
+#endif
+      this.createOutputViews();
+
+#if ASSERTIONS
+      // Explicitly verify this later in process()
+      this.ctorOldStackPtr = stackSave();
+#endif
+    }
+
+    /**
+     * Create up-front as many typed views for marshalling the output data as
+     * may be required (with an arbitrary maximum of 64, for the case where a
+     * multi-MB stack is passed), allocated at the *top* of the worklet's stack
+     * (and whose addresses are fixed). The 'minimum alloc' firstly stops
+     * STACK_OVERFLOW_CHECK failing (since the stack will be full, and 16 bytes
+     * being the minimum allocation size due to alignments) and leaves room for
+     * a single AudioSampleFrame as a minumum.
+     */
+    createOutputViews() {
+      // These are still alloc'd to take advantage of the overflow checks, etc.
+      var oldStackPtr = stackSave();
+      var viewDataIdx = {{{ getHeapOffset('stackAlloc(this.maxBuffers * this.bytesPerChannel)', 'float') }}};
+#if WEBAUDIO_DEBUG
+      console.log(`AudioWorklet creating ${this.maxBuffers} buffer one-time views (for a stack size of ${wwParams.stackSize} at address ${ptrToString(viewDataIdx * 4)})`);
+#endif
+      this.outputViews.length = 0;
```

Comment:
Maybe just have `this.outputViews.length = 0;` instead be `this.outputViews = [];`

It is an extremely rare occurrence that `createOutputViews()` needs to be called again. It will happen in an app lifetime maybe 10-20 times at most. (being generous). So there's no perf implication to always reassigning to an empty `= [];` from scratch and letting the old GC.